aggregator_core/src/datastore.rs

```diff
@@ -1497,11 +1528,13 @@ impl<C: Clock> Transaction<'_, C> {
                 // `None` (since all reads in the same transaction are from the same snapshot), so
                 // so it can't evaluate idempotency. All it can do is give up on this transaction
                 // and try again, by calling `retry` and returning an error; once it retries, it
-                // will be able to read the report written by the successful writer. (It doesn't
-                // matter what error we return here, as the transaction will be retried.)
+                // will be able to read the report written by the successful writer.
                 self.retry();
-                return Err(Error::MutationTargetAlreadyExists);
-            }
+                Error::DbState(
```

Comment:
It might be worth making a separate error variant for this sort of message, because `DbState` is currently used for unrecoverable corruption (i.e. wrong schema version, byte array of the wrong length, application-level invariant violated, etc.)